content/blog/2022/2022-04-26-introduction-to-retrofit.md

```diff
@@ -0,0 +1,536 @@
+
+## Setting up an existing REST Service Application
+We will use a sample REST-based [Spring Boot Library Application](https://github.com/ranjanih/code-examples/tree/ranjani-retrofit/core-java/retrofit/introduction-to-retrofit/SimpleLibraryApplication) that acts as a REST service.
+This library application is a Spring Boot service that uses Maven for build and HSQLDB as the underlying database.
+The Maven Wrapper bundled with the application will be used to start the service:
+````text
+    mvnw clean verify spring-boot:run (for Windows)
+    ./mvnw clean verify spring-boot:run (for Linux)
+````
+Now, the application should successfully start:
+{{% image alt="settings" src="images/posts/retrofit/success_at_startup.jpg" %}}
+
+The Swagger documentation of this application can be viewed at 
+````text
+    http://localhost:8090/swagger-ui.html
+````
+The documentation should look like this:
+{{% image alt="settings" src="images/posts/retrofit/swagger-doc.jpg" %}}
+
+Before we use swagger to make REST calls, we will add basic authentication credentials as configured in application.yaml.
+{{% image alt="settings" src="images/posts/retrofit/basic-auth.jpg" %}}
+
+Now, we can hit the REST endpoints successfully. (Sample JSON requests are available in README.md file in the application codebase.)
+{{% image alt="settings" src="images/posts/retrofit/POST.jpg" %}}
+{{% image alt="settings" src="images/posts/retrofit/POST-response.jpg" %}}
+
+Once the POST request is successful, we should be able to make a GET call to confirm this addition.
+{{% image alt="settings" src="images/posts/retrofit/GET-response.jpg" %}}
+
+Now that our REST service works as expected, we will move on to introduce another application that will act as a REST client making calls to this service.
+In the process, we will learn about Retrofit and its various features.
+
+## Introduction to REST Client Application
+The REST Client application will be a [Spring Boot Library Audit application]((https://github.com/ranjanih/code-examples/tree/ranjani-retrofit/core-java/retrofit/introduction-to-retrofit/AuditApplication)) that exposes REST endpoints and uses Retrofit to call our previously setup Library application. The result is then audited in an in-memory database for tracking purposes.
+
+## Adding Retrofit dependencies
+With `Maven`:
+````xml
+        <dependency>
+			<groupId>com.squareup.retrofit2</groupId>
+			<artifactId>retrofit</artifactId>
+			<version>2.5.0</version>
+		</dependency>
+		<dependency>
+			<groupId>com.squareup.retrofit2</groupId>
+			<artifactId>converter-jackson</artifactId>
+			<version>2.5.0</version>
+		</dependency>
+````
+With `Gradle`:
+````groovy
+        dependencies {  
+            implementation 'com.squareup.retrofit2:retrofit:2.5.0'
+            implementation 'com.squareup.retrofit2:converter-jackson:2.5.0'
+        }
+````
+
+## Setting Up a Retrofit Client
```

Comment:
It took a while until I understood that this section is about "quickly" showing what steps are needed. We should rename this section to "Quick Guide to Setting up a Retrofit Client" and mention in the introduction that we're going to look at the details in the upcoming section. I was confused as to why things were repeated in the next section.

Reply:
Makes sense. Updated.